core/tee/tee_svc_storage.c

```diff
@@ -710,7 +710,13 @@ TEE_Result syscall_storage_start_enum(unsigned long obj_enum,
 		return TEE_ERROR_ITEM_NOT_FOUND;
 
 	e->fops = fops;
-	assert(!e->dir);
+
+	if (e->fops && e->dir) {
```

Comment:
No need to check `e->fops`, it was just assigned something which was checked to be !NULL.